pkg/container-collection/k8s.go

```diff
@@ -106,6 +107,7 @@ func (k *K8sClient) PodToContainers(pod *v1.Pod) []Container {
 
 	containerStatuses := append([]v1.ContainerStatus{}, pod.Status.InitContainerStatuses...)
 	containerStatuses = append(containerStatuses, pod.Status.ContainerStatuses...)
+	containerStatuses = append(containerStatuses, pod.Status.EphemeralContainerStatuses...)
```

Comment:
The field is defined as follows:
```
// Status for any ephemeral containers that have run in this pod.
// +optional
EphemeralContainerStatuses []ContainerStatus `json:"ephemeralContainerStatuses,omitempty" protobuf:"bytes,13,rep,name=ephemeralContainerStatuses"`
```

I have not tested an old Kubernetes version but I don't think it would error out. Inspektor Gadget is compiled with k8s.io/api v0.25.4 (see go.mod) so the field `EphemeralContainerStatuses` exists. The API server might not know about that field so it would send a PodStatus without it, so the PodStatus json text would be unmarshalled in Inspektor Gadget as an empty array, but that's ok. In any case, this patch does not change the unmarshalling; this specific line does not call any function that calls the API server, so it cannot error out.

Reply:
It won't show any compilation error because we compile against a version that contains those fields.
I suppose the server won't return those fields and they will be nil in the client, hence I don't think we need any version check there.

Reply:
(of course I commented without seeing Alban's answer that is a lot more complete than mine).